homeassistant/components/bmw_connected_drive/binary_sensor.py

```diff
@@ -16,89 +19,101 @@
-}
-
-SENSOR_TYPES_ELEC.update(SENSOR_TYPES)
+SENSOR_TYPES: tuple[BinarySensorEntityDescription, ...] = (
+    BinarySensorEntityDescription(
+        key="lids",
+        name="Doors",
+        device_class=DEVICE_CLASS_OPENING,
+        icon="mdi:car-door-lock",
+    ),
+    BinarySensorEntityDescription(
+        key="windows",
+        name="Windows",
+        device_class=DEVICE_CLASS_OPENING,
+        icon="mdi:car-door",
+    ),
+    BinarySensorEntityDescription(
+        key="door_lock_state",
+        name="Door lock state",
+        device_class="lock",
+        icon="mdi:car-key",
+    ),
+    BinarySensorEntityDescription(
+        key="lights_parking",
+        name="Parking lights",
+        device_class="light",
+        icon="mdi:car-parking-lights",
+    ),
+    BinarySensorEntityDescription(
+        key="condition_based_services",
+        name="Condition based services",
+        device_class=DEVICE_CLASS_PROBLEM,
+        icon="mdi:wrench",
+    ),
+    BinarySensorEntityDescription(
+        key="check_control_messages",
+        name="Control messages",
+        device_class=DEVICE_CLASS_PROBLEM,
+        icon="mdi:car-tire-alert",
+    ),
+    # electric
+    BinarySensorEntityDescription(
+        key="charging_status",
+        name="Charging status",
+        device_class="power",
+        icon="mdi:ev-station",
+    ),
+    BinarySensorEntityDescription(
+        key="connection_status",
+        name="Connection status",
+        device_class=DEVICE_CLASS_PLUG,
+        icon="mdi:car-electric",
+    ),
+)
 
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the BMW ConnectedDrive binary sensors from config entry."""
     account = hass.data[BMW_DOMAIN][DATA_ENTRIES][config_entry.entry_id][CONF_ACCOUNT]
-    entities = []
-
-    for vehicle in account.account.vehicles:
-        if vehicle.has_hv_battery:
-            _LOGGER.debug("BMW with a high voltage battery")
-            for key, value in sorted(SENSOR_TYPES_ELEC.items()):
-                if key in vehicle.available_attributes:
-                    device = BMWConnectedDriveSensor(
-                        account, vehicle, key, value[0], value[1], value[2]
-                    )
-                    entities.append(device)
-        elif vehicle.has_internal_combustion_engine:
-            _LOGGER.debug("BMW with an internal combustion engine")
-            for key, value in sorted(SENSOR_TYPES.items()):
-                if key in vehicle.available_attributes:
-                    device = BMWConnectedDriveSensor(
-                        account, vehicle, key, value[0], value[1], value[2]
-                    )
-                    entities.append(device)
+
+    entities = [
+        BMWConnectedDriveSensor(account, vehicle, description)
+        for vehicle in account.account.vehicles
+        for description in SENSOR_TYPES
+        if description.key in vehicle.available_attributes
+    ]
```

Comment:
Keys are only added to `available_attributes` based on the vehicle type in the first place. Thus it's safe to combine them as an internal combustion engine car will never have the `charging_status` attribute.
[bimmer_connected/vehicle.py](https://github.com/bimmerconnected/bimmer_connected/blob/cea6cd8b1e463e6b47b875ea09689914bc7f28a2/bimmer_connected/vehicle.py#L140-L161)